### PR TITLE
Sparse gradient add

### PR DESCRIPTION
**Pull request by zkh2016**
### PR types

Others
### PR changes

Others
### Describe

GradTensorHolder::add添加对SparseCooTensor梯度累加的支持

### CODE REVIEW

File: paddle/fluid/eager/grad_tensor_holder.cc [hunk 1]
```diff
@@ -130,6 +131,26 @@ void GradTensorHolder::add(size_t slot_id, size_t rank,
                                                   &new_buffer);
         buffer_tensor.set_impl(new_buffer.impl());
       }
+    } else if (t.is_sparse_coo_tensor()) {
```
Comment (JiabinYang) [inline paddle/fluid/eager/grad_tensor_holder.cc:134]: support this is accumulation_node.cc

Comment (JiabinYang) [reply paddle/fluid/eager/grad_tensor_holder.cc:134]: and how about dense add to sparse?

**Review by JiabinYang — COMMENTED** [review]
some comment

File: paddle/fluid/eager/grad_tensor_holder.cc (hunk 1 above)
Comment (zkh2016) [reply paddle/fluid/eager/grad_tensor_holder.cc:134]: > support this is accumulation_node.cc

done

> and how about dense add to sparse?

The sparse API requires that the gradient of SparseTensor is still a SparseTenosr, so dense+sparse is not required at present.

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**